api/src/main/java/uk/gov/dhsc/htbhf/dwp/service/HouseholdVerifier.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.dhsc.htbhf.dwp.service;
+
+import org.flywaydb.core.internal.util.StringUtils;
+import org.springframework.stereotype.Service;
+import uk.gov.dhsc.htbhf.dwp.entity.legacy.LegacyHousehold;
+import uk.gov.dhsc.htbhf.dwp.entity.uc.UCHousehold;
+import uk.gov.dhsc.htbhf.dwp.model.PersonDTO;
+
+@Service
+public class HouseholdVerifier {
+
+    /**
+     * Determines if a given household contains an adult matching a given person.
+     * DWP rules specify to only match on the first six characters of the address line 1 field.
```

Comment:
This documentation is a bit confusing as you only specify the address match, but in fact it matches on much more, can you detail all the matches it does here please?

Reply:
The description in the javadoc refers specifically to the matching for an address, but we also match on name.

Reply:
I think we'll want to be quite explicit on how we match on name & address in the review/showcase. I'm a little worried that first names may not be so easy to match - Matt vs Matthew, etc.

Reply:
Updated to include all the checks we perform.